shopsite/shop/UserAuthorization.py

```diff
@@ -0,0 +1,127 @@
+from tastypie.authorization import Authorization
+from tastypie.exceptions import Unauthorized
+from .models import User, Product, Order, Category
+import waffle
+
+
+class UserAuthorization(Authorization):
+    def read_list(self, object_list, bundle):
+        model_class = object_list.model
+
+        if model_class == User:
+            if not waffle.flag_is_active(bundle.request, 'isSuperUser'):
```

Comment:
Actually, django user has field `is_superuser`, it's not necessary to create a separate waffle flag for it.